src/core/server/status/status_service.ts

```diff
@@ -18,39 +18,65 @@
  */
 
 import { Observable, combineLatest } from 'rxjs';
-import { map, distinctUntilChanged, shareReplay, take } from 'rxjs/operators';
+import { map, distinctUntilChanged, shareReplay, take, debounce } from 'rxjs/operators';
 import { isDeepStrictEqual } from 'util';
 
 import { CoreService } from '../../types';
 import { CoreContext } from '../core_context';
 import { Logger } from '../logging';
 import { InternalElasticsearchServiceSetup } from '../elasticsearch';
 import { InternalSavedObjectsServiceSetup } from '../saved_objects';
+import { PluginName } from '../plugins';
 
 import { config, StatusConfigType } from './status_config';
 import { ServiceStatus, CoreStatus, InternalStatusServiceSetup } from './types';
 import { getSummaryStatus } from './get_summary_status';
+import { PluginsStatusService } from './plugins_status';
 
 interface SetupDeps {
   elasticsearch: Pick<InternalElasticsearchServiceSetup, 'status$'>;
+  pluginDependencies: ReadonlyMap<PluginName, PluginName[]>;
   savedObjects: Pick<InternalSavedObjectsServiceSetup, 'status$'>;
 }
 
+const waitForMicrotasks = (numberOfTicks: number = 1): Promise<unknown> => {
+  if (numberOfTicks === 1) {
+    return Promise.resolve();
+  } else {
+    return Promise.resolve().then(() => waitForMicrotasks(numberOfTicks - 1));
+  }
+};
+
 export class StatusService implements CoreService<InternalStatusServiceSetup> {
   private readonly logger: Logger;
   private readonly config$: Observable<StatusConfigType>;
 
+  private pluginsStatus?: PluginsStatusService;
+
   constructor(coreContext: CoreContext) {
     this.logger = coreContext.logger.get('status');
     this.config$ = coreContext.configService.atPath<StatusConfigType>(config.path);
   }
 
-  public async setup(core: SetupDeps) {
+  public async setup({ elasticsearch, pluginDependencies, savedObjects }: SetupDeps) {
     const statusConfig = await this.config$.pipe(take(1)).toPromise();
-    const core$ = this.setupCoreStatus(core);
-    const overall$: Observable<ServiceStatus> = core$.pipe(
-      map((coreStatus) => {
-        const summary = getSummaryStatus(Object.entries(coreStatus));
+    const core$ = this.setupCoreStatus({ elasticsearch, savedObjects });
+    this.pluginsStatus = new PluginsStatusService({ core$, pluginDependencies });
+
+    const overall$: Observable<ServiceStatus> = combineLatest(
+      core$,
+      this.pluginsStatus.getAll$()
+    ).pipe(
+      // We schedule the number of microtasks expected to resolve all of the dependencies of this update.
+      // Waiting for this ensures that we do not emit 'partial updates' to reduce noise.
+      debounce(([coreStatus, pluginsStatus]) =>
+        waitForMicrotasks(1 + Object.keys(pluginsStatus).length)
```

Comment:
does it mean that all `this.pluginsStatus` API consumers have to implement `debouncing` manually?

Reply:
Now that debouncing is configured on the Observable used to wrap a plugin's dependencies (the `core.status.dependencies$` Observable), I don't believe this will be a problem that "fans out" to consumers, so manual debouncing should be unnecessary.